File: app/src/main/java/ru/gb/course1/notebook/domain/Note.java
package ru.gb.course1.notebook.domain;

import android.os.Parcel;
import android.os.Parcelable;

public class Note implements Parcelable {
    private int idNote;
    private String title;
    private String description;
    private String timeCreate;

    public Note(int idNote, String title, String description, String timeCreate){
        this.idNote = idNote;
        this.title = title;
        this.description = description;
        this.timeCreate = timeCreate;
    }

    public Note(String title, String description) {
        this.title = title;
        this.description = description;
    }

    public Note(Parcel in) {
        idNote = in.readInt();
        title = in.readString();
        description = in.readString();
        timeCreate = in.readString();
    }

    public static final Creator<Note> CREATOR = new Creator<Note>() {
        @Override
        public Note createFromParcel(Parcel in) {
            return new Note(in);
        }

        @Override
        public Note[] newArray(int size) {
            return new Note[size];
        }
    };

    public void setIdNote(int idNote) {
        this.idNote = idNote;
    }

    public void setTimeCreate(String timeCreate) {
        this.timeCreate = timeCreate;
    }

    public String getTimeCreate() {
        return timeCreate;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public int getIdNote() {
        return idNote;
    }

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel parcel, int i) {
        parcel.writeInt(idNote);
        parcel.writeString(title);
        parcel.writeString(description);
        parcel.writeString(timeCreate);
    }
}

File: app/src/main/java/ru/gb/course1/notebook/domain/Note.kt
package ru.gb.course1.notebook.domain

import android.os.Parcelable
import android.os.Parcel
import android.os.Parcelable.Creator

class Note : Parcelable {
    var idNote = 0
    var title: String?
    var description: String?
    var timeCreate: String? = null

    constructor(idNote: Int, title: String?, description: String?, timeCreate: String?) {
        this.idNote = idNote
        this.title = title
        this.description = description
        this.timeCreate = timeCreate
    }

    constructor(title: String?, description: String?) {
        this.title = title
        this.description = description
    }

    constructor(`in`: Parcel) {
        idNote = `in`.readInt()
        title = `in`.readString()
        description = `in`.readString()
        timeCreate = `in`.readString()
    }

    override fun describeContents(): Int {
        return 0
    }

    override fun writeToParcel(parcel: Parcel, i: Int) {
        parcel.writeInt(idNote)
        parcel.writeString(title)
        parcel.writeString(description)
        parcel.writeString(timeCreate)
    }

    companion object CREATOR : Creator<Note> {
        override fun createFromParcel(parcel: Parcel): Note {
            return Note(parcel)
        }

        override fun newArray(size: Int): Array<Note?> {
            return arrayOfNulls(size)
        }
    }
}

